instrumentation/github.com/Shopify/sarama/otelsarama/producer.go

```diff
@@ -121,7 +121,7 @@ func (p *asyncProducer) Close() error {
 }
 
 type producerMessageContext struct {
-	span           trace.Span
+	span           *producerSpan
```

Comment:
Aby reason why a pointer is used? It is especially strange as the method does not handle nil value.

Reply:
Not really, just standard. I made it not a pointer as it doesn't really hurt copying the whole struct since there is only one field in it.

Reply:
BTW using values instead of pointers is often faster. E.g. take a look at https://medium.com/a-journey-with-go/go-should-i-use-a-pointer-instead-of-a-copy-of-my-struct-44b43b104963